docs/code-examples/text_log_integration.rs

```diff
@@ -0,0 +1,26 @@
+//! Shows integration of Rerun's `TextLog` with the native logging interface.
+
+use rerun::{archetypes::TextLog, components::TextLogLevel, external::log, RecordingStreamBuilder};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let (rec, storage) =
+        RecordingStreamBuilder::new("rerun_example_text_log_integration").memory()?;
+
+    // Log a text entry directly:
+    rec.log(
+        "logs",
+        &TextLog::new("this entry has loglevel TRACE").with_level(TextLogLevel::TRACE),
+    )?;
+
+    // Or log via a logging handler:
+    rerun::Logger::new(rec.clone()) // recording streams are ref-counted
+        .with_path_prefix("logs/handler")
+        // You can also use the standard `RUST_LOG` environment variable!
+        .with_filter(rerun::default_log_filter())
+        .init()?;
+    log::info!("This INFO log got added through the standard logging interface");
+
+    rerun::native_viewer::show(storage.take())?;
+    rec.disconnect(); // force-close the underlying sink to flush it
```

Comment:
That one cost me some hair...

The global logger singleton holds a reference to the recording stream, meaning the recording will never be dropped.
Surely this is fine in and of itself: those tests are run with `RERUN_FLUSH_NUM_ROWS`, so the recording being dropped or not doesn't change anything... and indeed, if you run this example manually, everything works fine.

_**But**_, when this runs in roundtrip mode, then suddenly the underlying sink is a file sink... and file sinks' flushes are completely decorrelated from recordings' flushes.
I.e. the roundtrip will fail, because the recording wasn't dropped, which means the file sink wasn't dropped, which means the file sink wasn't flushed (even though the recording was, thanks to `RERUN_FLUSH_NUM_ROWS=0`).

Reply:
> file sinks' flushes are completely decorrelated from recordings' flushes

This is the critical part, and that goes beyond log integration. Anytime a user clones a RecordingStream (knowingly or not), they expose themselves to flushing races when using file sinks due to this.
Let me see if I can do something quick about this...

Reply:
- https://github.com/rerun-io/rerun/pull/3525